components/scream/src/share/tests/field_tests.cpp

```diff
@@ -186,6 +186,28 @@ TEST_CASE("field", "") {
     REQUIRE(views_are_equal(f1,f2));
   }
 
+  SECTION ("clone") {
+    Field f1 (fid);
+    auto& fap1 = f1.get_header().get_alloc_properties();
+
+    fap1.request_allocation(16);
+    f1.allocate_view();
+    f1.deep_copy(3.0);
+
+    Field f2 = f1.clone();
+    auto& fap2 = f2.get_header().get_alloc_properties();
+    REQUIRE(f2.is_allocated());
+    REQUIRE(fap2.get_alloc_size()==fap1.get_alloc_size());
+    REQUIRE(views_are_equal(f1,f2));
+
+    // Changing f2 should leave f1 unchanged
```

Comment:
Is it guaranteed that changing f1 similarly leaves f2 unchanged?

Reply:
I think so.